src/DittoMachine.sol

```diff
@@ -60,80 +61,80 @@ contract DittoMachine is ERC721, ERC721TokenReceiver {
+            ));
+
+            string memory image = Base64.encode(bytes(generateSVGofTokenById(id)));
+
+            return string(abi.encodePacked(
+                'data:application/json;base64,',
+                Base64.encode(
+                    bytes(
+                        abi.encodePacked(
+                           '{"name":"',
+                            _name,
+                           '", "description":"',
+                           description,
+                           '", "attributes": [{"trait_type": "Underlying NFT", "value": "',
+                           Strings.toHexString(uint160(cloneIdToShape[id].ERC721Contract), 20),
+                           '"},{"trait_type": "tokenId", "value": ',
+                           Strings.toString(cloneShape.tokenId),
+                           '}], "owner":"',
+                           Strings.toHexString(uint160(ownerOf[id]), 20),
+                           '", "image": "',
+                           'data:image/svg+xml;base64,',
+                           image,
+                           '"}'
+                        )
                     )
                 )
-            )
-        ));
-    }
-
-    // Visibility is `public` to enable it being called by other contracts for composition.
-    function renderTokenById(uint256 id) public view returns (string memory) {
-        CloneShape memory cloneShape = cloneIdToShape[id];
-        string memory nftTokenId = cloneShape.floor ? "Floor" : Strings.toString(cloneShape.tokenId);
-
-        string memory render = string(abi.encodePacked(
-            '<g>',
-                '<style>',
-                    '.small { font: italic 13px sans-serif; }',
-                    '.Rrrrr { font: italic 40px serif; fill: red; }',
-                '</style>',
-                '<text x="20" y="35" class="small">',Strings.toHexString(uint160(cloneIdToShape[id].ERC721Contract), 20),'</text>',
-                '<text x="55" y="65" class="Rrrrr">',ERC721(cloneShape.ERC721Contract).name(),'</text>',
-                '<text x="250" y="70" class="small">',nftTokenId,'</text>',
-            '</g>'
-        ));
-
-      return render;
+            ));
+        }
     }
 
-    function generateSVGofTokenById(uint256 id) internal view returns (string memory) {
-
+    function generateSVGofTokenById(uint256 _tokenId) internal pure returns (string memory) {
         string memory svg = string(abi.encodePacked(
-          '<svg viewBox="0 0 340 310" xmlns="http://www.w3.org/2000/svg">',
-            renderTokenById(id),
+          '<svg viewBox="0 0 400 400" xmlns="http://www.w3.org/2000/svg">',
+            renderTokenById(_tokenId),
           '</svg>'
         ));
 
         return svg;
     }
 
+    // Visibility is `public` to enable it being called by other contracts for composition.
+    function renderTokenById(uint256 _tokenId) public pure returns (string memory) {
```

Comment:
We are trying to push for these semantics, as its easier for others to take the core SVG and plug it in theirs.